bin/ch/WScriptJsrt.cpp

```diff
@@ -515,13 +515,24 @@ bool WScriptJsrt::PrintException(LPCWSTR fileName, JsErrorCode jsErrorCode)
             {
                 JsPropertyIdRef stackPropertyId = JS_INVALID_REFERENCE;
                 JsValueRef stackProperty = JS_INVALID_REFERENCE;
+                JsValueRef errorString = JS_INVALID_REFERENCE;
+                JsValueType stackPropertyType;
                 LPCWSTR errorStack = nullptr;
                 size_t errorStackLength = 0;
 
                 IfJsrtErrorFail(ChakraRTInterface::JsGetPropertyIdFromName(L"stack", &stackPropertyId), false);
                 IfJsrtErrorFail(ChakraRTInterface::JsGetProperty(exception, stackPropertyId, &stackProperty), false);
-                IfJsrtErrorFail(ChakraRTInterface::JsStringToPointer(stackProperty, &errorStack, &errorStackLength), false);
+                IfJsrtErrorFail(ChakraRTInterface::JsGetValueType(stackProperty, &stackPropertyType), false);
 
+                if (stackPropertyType == JsUndefined)
+                {
+                    IfJsrtErrorFail(ChakraRTInterface::JsConvertValueToString(exception, &errorString), false);
```

Comment:
You could refactor some of the code here. Lines between `493` and `495` are extracting out message from exception object. `ConvertValueToString` also extracts `message` and `name` from exception object. If we refactor this, we should also be changing around line 512.

I have a prototype ready of the refactoring I am proposing and if you want i can send a PR for this. Let me know.